cesium/internal/domain/index.go

```diff
@@ -34,13 +35,17 @@ type indexUpdate struct {
 var _ observe.Observable[indexUpdate] = (*index)(nil)
 
 // insert adds a new pointer to the index.
-func (idx *index) insert(ctx context.Context, p pointer) error {
+func (idx *index) insert(ctx context.Context, p pointer, persist bool) error {
 	_, span := idx.T.Bench(ctx, "insert")
+	idx.mu.Lock()
+
 	defer span.End()
-	idx.mu.RLock()
+
 	insertAt := 0
+
 	if p.fileKey == 0 {
-		panic("fileKey must be set")
+		idx.mu.Unlock()
+		idx.L.DPanic("fileKey must be set")
```

Comment:
Remember that this will only panic in development. In production this function will continue execution. Are you sure you don't want to return here?